### PR TITLE
Excel create sheet

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/612)

### CODE REVIEW

File: reporter/src/main/kotlin/reporters/ExcelReporter.kt [hunk 1]
```diff
@@ -45,20 +45,20 @@ import java.io.File
  * for each project in [ScanRecord.analyzerResult] and an additional sheet that summarizes all dependencies.
  */
 class ExcelReporter : TableReporter() {
-    val DEFAULT_COLUMNS = 5
+    private val defaultColumn = 5
```
Comment (mnonnenmacher) [inline reporter/src/main/kotlin/reporters/ExcelReporter.kt]: This seems like a broken inspection, it should actually propose to use `const val DEFAULT_COLUMNS = 5` as in the style guide:
https://kotlinlang.org/docs/reference/coding-conventions.html#property-names

It seems to propose to use `const` for `FIELDS` below, weird.

Also it should be plural `defaultColumns`.

Comment (sschuberth) [reply reporter/src/main/kotlin/reporters/ExcelReporter.kt]: Unfortunately, "Const 'val' are only allowed on top level or in objects".

> Also it should be plural `defaultColumns`.

Fixed that!

File: reporter/src/main/kotlin/reporters/ExcelReporter.kt [hunk 2]
```diff
@@ -244,15 +244,18 @@ class ExcelReporter : TableReporter() {
...
+        WorkbookUtil.createSafeSheetName(name).let { safeName ->
+            var uniqueName = safeName
             var i = 0
 
             while (uniqueName in Sequence { workbook.sheetIterator() }.map { it.sheetName }) {
                 val suffix = "-${++i}"
-                uniqueName = uniqueName.dropLast(suffix.length) + suffix
+                uniqueName = uniqueName.take(MAX_SHEET_NAME_LENGTH - suffix.length) + suffix
             }
+
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply reporter/src/main/kotlin/reporters/ExcelReporter.kt]: Your original code to append the suffix makes the name exceed 31 chars if `WorkbookUtil.createSafeSheetName()` already returned a name with 31 chars.

[the remark this answers is not in the document]
Comment (sschuberth) [reply reporter/src/main/kotlin/reporters/ExcelReporter.kt]: Hmm, scratch that. Your original code *should* work. Let me think about it again.

[the remark this answers is not in the document]
Comment (sschuberth) [reply reporter/src/main/kotlin/reporters/ExcelReporter.kt]: I've just added a test for now.

**Pull request merged**